pilot/pkg/config/kube/crdclient/client.go

```diff
@@ -421,31 +421,8 @@ func handleCRDAdd(cl *Client, name string) {
 		cl.logger.Debugf("added resource that already exists: %v", resourceGVK)
 		return
 	}
-	var i informers.GenericInformer
-	var ifactory starter
-	var err error
-	switch s.Group() {
-	case gvk.KubernetesGateway.Group:
-		ifactory = cl.client.GatewayAPIInformer()
-		i, err = cl.client.GatewayAPIInformer().ForResource(gvr)
-	case gvk.Pod.Group, gvk.Deployment.Group, gvk.MutatingWebhookConfiguration.Group:
-		ifactory = cl.client.KubeInformer()
-		i, err = cl.client.KubeInformer().ForResource(gvr)
-	case gvk.CustomResourceDefinition.Group:
-		ifactory = cl.client.ExtInformer()
-		i, err = cl.client.ExtInformer().ForResource(gvr)
-	default:
-		ifactory = cl.client.IstioInformer()
-		i, err = cl.client.IstioInformer().ForResource(gvr)
-	}
-	if err != nil {
-		// Shouldn't happen
-		cl.logger.Errorf("failed to create informer for %v: %v", resourceGVK, err)
-		return
-	}
-	_ = i.Informer().SetTransform(kube.StripUnusedFields)
-
-	cl.kinds[resourceGVK] = createCacheHandler(cl, s, i)
+	inf := kclient.NewUntyped(cl.client, gvr, kclient.Filter{ObjectFilter: cl.namespacesFilter})
```

Comment:
Suggest rnaming NewUntyped to NewUntypedInformer.

NewUntyped is called everywhere, and it makes me think it is to create a client for a specific type